libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1326,6 +1329,13 @@ class MapboxRouteLineApi(
         )
     }
 
+    private fun resetCaches() {
+        MapboxRouteLineUtils.resetCache()
+        repeat(ALTERNATIVELY_STYLE_SEGMENTS_CACHE_SIZE) {
```

Comment:
Isn't there a more idiomatic way to clear the LRU cache instead of overwriting the data?

Reply:
I can see https://developer.android.com/reference/android/util/LruCache#evictAll(), can we take advantage of that?

Reply:
The cache is embedded in the extension function it's not accessible from the outside. This is the next best thing I could come up with. It accomplishes the task that was asked for, it removes the routes from the cache.

Reply:
@cafesilencio Great catch with route line cache issue. I was wondering why route lines are not clearing after calling `mapboxNavigation.setRoutes(emptyList())`.

What do you guys thing about making `CacheResultUtils.CacheResultHandler` internal, adding `clearCache()` method...
```kotlin
internal class CacheResultHandler<F, in K : CacheResultCall<F, R>, out R>(
    val f: F,
    maxSize: Int
) {
    private val cache = LruCache<K, R>(maxSize)
    operator fun invoke(k: K): R {
        synchronized(cache) {
            return cache[k] ?: run {
                val r = k(f)
                if (cache.get(k) == null) {
                    cache.put(k, r)
                }
                r
            }
        }
    }

    fun clearCache() = cache.evictAll()
}
```
...and calling it instead?